dolphinscheduler-task-plugin/dolphinscheduler-task-sql/src/main/java/org/apache/dolphinscheduler/plugin/task/sql/SqlTask.java

```diff
@@ -197,7 +198,9 @@ public void executeFuncAndSql(List<SqlBinds> mainStatementsBinds,
             // decide whether to executeQuery or executeUpdate based on sqlType
             if (sqlParameters.getSqlType() == SqlType.QUERY.ordinal()) {
                 // query statements need to be convert to JsonArray and inserted into Alert to send
-                resultSet = executeQuery(connection, mainStatementsBinds.get(0), "main");
+                statement = prepareStatementAndBind(connection, mainStatementsBinds.get(0));
+                logger.info("main statement execute query, for sql: {}", mainStatementsBinds.get(0).getSql());
+                resultSet = statement.executeQuery();
```

Comment:
I do not recommend this modification. You can put the resultProcess method in the query method, and the close method only has connection as an input parameter.

Reply:
Its better to put the resultProcess method in the query method, i will fix my code.

Reply:
the close method only has connection as an input parameter, it is also recommended to modify.

Reply:
resultSet is allways null， remove from the close method